packages/critters/src/index.js

```diff
@@ -547,10 +547,16 @@ export default class Critters {
             }
             sel = sel
               .replace(/(?<!\\)::?[a-z-]+(?![a-z-(])/gi, '')
-              .replace(/::?not\(\s*\)/g, '')
```

Comment:
Should this be removed? This removes the case for :not() selector.

Reply:
The code that replaced it handles any `::<something>()` empty or with spaces only

Reply:
In a better breakdown (sorry if I'm explaining the basics, it's just so that we get aligned on it, I'm not assuming any knowledge-level 😉 ):

For the RegExp `/(::?[a-z0-9-]+)\(([^)]*)\)/gi`:

1. The first group - `(::?[a-z0-9-]+)` - captures cases like `:not` or `::part`
2. The second group - `([^)]*)` - captures what is inside of the parenthesis.
3. The `gi` flags capture globally and case insensitively (though the case doesn't really matter)

With that said, the replace function

```js
              (_, p1, p2) => {
                const params = (p2 || '')
                  .split(',')
                  .map((item) => item.trim())
                  .filter(Boolean)
                  .join(',');

                return params ? `${p1}(${params})` : '';
              }
```

Takes whatever it is inside the parenthesis - e.g. in `:not(.myClass,,,.myClass2)` it'll get the string `.myClass,,,.myClass2` - and splits the values by `,`, maps through them, trimming them (guaranteeing that the values do not have left and right spaces), then filters only the truthy values, and then joins them back with ',' ... Following the example the variable `params` will contain the string `.myClass,.myClass2`.

Finally, if the variable `params` contains values (could be empty) it returns the group 1 of the RegExp + the string from `params` wrapped in parenthesis.
Following the example it'd be `:not(.myClass,.myClass2)`
If `params` is empty, returns empty string replacing the whole pseudo-selector.

Would you consider this good enough to be merged @janicklas-ralph ?

Reply:
Should I close this? We've been patching Critters with this code so far in our CI and didn't have any problems ... 🤔